### PR TITLE
Propagate the measure label when using create_metric option

### LINKED ISSUE #10538: [Bug] Measure labels for create_metric = true don't propogate
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

When using create_metric on a measure with a label defined, the metric label is the measure name

### Expected Behavior

The metric label should be the measure label according to https://docs.getdbt.com/docs/build/measures#:~:text=The%20label%20and%20description%20attributes%20will%20be%20automatically%20propagated%20to%20the%20created%20metric.

### Steps To Reproduce

Add create_metric on a measure with a label

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt:
```

### Which database adapter are you using with dbt?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by aliceliu**
Resolves #10538 

### Problem

When using create_metric on a measure, the label of the measure is not propagated to the metric

### Solution

Set metric label to be measure label if it exists

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me.
- [x] I have run this code in development, and it appears to resolve the stated issue.
- [x] This PR includes tests, or tests are not required or relevant for this PR.
- [x] This PR has no interface changes (e.g., macros, CLI, logs, JSON artifacts, config files, adapter interface, etc.) or this PR has already received feedback and approval from Product or DX.
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions.

### CODE REVIEW

**Comment by aliceliu** [comment]
* **#10536** <a href="https://app.graphite.dev/github/pr/dbt-labs/dbt-core/10536?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @aliceliu and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by WilliamDee — APPROVED** [review]

**Review by QMalcolm — APPROVED** [review]
This looks good to me. It would probably be worthwhile to add a test which checks the state of the created metric when the related measure _does not_ have a label set, however this is not blocking.

## Context
When `_create_metric`, which creates a metric from a measure, was written, measures did not have labels. It seems that when `label` was added to the measure class, we did not think to add it here. That is an oversight, but I can see how that happened.

## Implications

For metrics created from measures, in order to get the correct label, a _full reparse_ will be necessary. This is because partial parsing only reparses an object is the _input_ for the object has changed. Thus, without a full parsing, the relevant metrics will only get updated if the underlying semantic model is modified somehow.

### TL;DR
Looks good. People will need to run `dbt parse --no-partial-parse` to ensure all _existing_ measures with labels that create metrics properly propagate the label.

**Comment by aliceliu** [comment]
@QMalcolm thanks for the context! I updated the test. I don't have permission to merge so I'll leave the merging to you

**Comment by Vitor-Avila** [comment]
Hey @QMalcolm, @WilliamDee 
I have a question about this issue. I've created a measure like below:
``` yaml
measures:
  - name: metric_measure
    description: This is a measure (originally)
    label: This is the label value
    expr: order_id
    agg: count_distinct
    create_metric: true
```

After running both `dbt run` and/or `dbt parse --no-partial-parse`, I see the metric declared in the `manifest.json` like this:
``` json
{
    ...
    "metric.vec_sales.metric_measure":
    {
        "name": "metric_measure",
        "resource_type": "metric",
        "package_name": "vec_sales",
        "path": "vehicle_sales.yml",
        "original_file_path": "models/vehicle_sales.yml",
        "unique_id": "metric.vec_sales.metric_measure",
        "fqn": ["vec_sales", "metric_measure"],
        "description": "This is a measure (originally)",
        "label": "metric_measure",
        ...
}
```

The `label` in the `manifest.json` is still incorrectly using the measure name. Should I submit a new ticket about this? Am I missing something?

``` bash
(dbt-latest) ➜  vec_sales pip freeze | grep dbt
dbt-adapters==1.10.3
dbt-common==1.13.0
dbt-core==1.8.9
dbt-extractor==0.5.1
dbt-postgres==1.8.2
dbt-semantic-interfaces==0.5.1
```

**Comment by QMalcolm** [comment]
@Vitor-Avila howdy! Looks like this never got backported to 1.8.latest and thus never went into a 1.8.x minor release. I can check in with the team if that is something we want to do, it shouldn't be a major lift.

However, we also plan to release 1.9.0 very soon. It should be fixed in that version. If you want a sneak peak of that, try installing the latest release candidate via `pip install dbt-core==1.9.0rc2` (which should resolve your issue)

**Comment by Vitor-Avila** [comment]
Thank you so much for the quick response, @QMalcolm! 🙏 I've just tested using `1.9.0rc2` and confirmed it works. Appreciate the assistance here.
